Wrappers/Python/cil/io/TXRMTXMDataReader.py

```diff
@@ -100,46 +128,59 @@ def read(self):
                     'ImageInfo/DtoRADistance', "<{0}f".format(number_of_images))
             
-        dist_source_center   = np.abs( StoRADistance[0] )
-        dist_center_detector = np.abs( DtoRADistance[0] )
-        
-        # normalise data by flatfield
-        data = data / metadata['reference']
-        
-        # circularly shift data by rounded x and y shifts
-        for k in range(number_of_images):
-            data[k,:,:] = np.roll(data[k,:,:], \
-                (int(metadata['x-shifts'][k]),int(metadata['y-shifts'][k])), \
-                axis=(1,0))
-        
-        # Pixelsize loaded in metadata is really the voxel size in um.
-        # We can compute the effective detector pixel size as the geometric
-        # magnification times the voxel size.
-        d_pixel_size = ((dist_source_center+dist_center_detector)/dist_source_center)*metadata['pixel_size']
-        
-        # convert angles to requested unit measure, Zeiss stores in radians
-        if self.angle_unit == AcquisitionGeometry.DEGREE:
-            angles = np.degrees(metadata['thetas'])
-        else:
-            angles = np.asarray(metadata['thetas'])
+            dist_source_center   = np.abs( StoRADistance[0] )
+            dist_center_detector = np.abs( DtoRADistance[0] )
 
-        self._ag = AcquisitionGeometry.create_Cone3D(
-            [0,-dist_source_center, 0] , [ 0, dist_center_detector, 0] \
-            ) \
-                .set_panel([metadata['image_width'], metadata['image_height']],\
-                    pixel_size=[d_pixel_size/1000,d_pixel_size/1000])\
-                .set_angles(angles, angle_unit=self.angle_unit)
-        self._ag.dimension_labels =  ['angle', 'vertical', 'horizontal']
+        if dist_source_center != dist_center_detector:
+            # Pixelsize loaded in metadata is really the voxel size in um.
+            # We can compute the effective detector pixel size as the geometric
+            # magnification times the voxel size.
+            d_pixel_size = ((dist_source_center+dist_center_detector)/dist_source_center)*metadata['pixel_size']
                 
+            self._ag = AcquisitionGeometry.create_Cone3D(
+                [0,-dist_source_center, 0] , [ 0, dist_center_detector, 0] \
+                ) \
+                    .set_panel([metadata['image_width'], metadata['image_height']],\
+                        pixel_size=[d_pixel_size/1000,d_pixel_size/1000])\
+                    .set_angles(angles[image_range],angle_unit=self.angle_unit)
+        else:
+            self._ag = AcquisitionGeometry.create_Parallel3D()\
+                        .set_panel([metadata['image_width'], metadata['image_height']])\
+                        .set_angles(angles[image_range],angle_unit=self.angle_unit)
+        self._ag.dimension_labels =  ['angle', 'vertical', 'horizontal']
+
+        ext = os.path.splitext(self.txrm_file)
+        if ext[1] == '.txrm':
```

Comment:
This check should happen initially. If the file contains `AcquisitionData` then it should read it as such; likewise for `ImageData`. 
I'm not sure the extension check is sufficient here? Maybe we could make the `get_metadata` smarter and understand if we are dealing with acquisition or image data.

The point here is that this check happens after we've tried to read the file as `AcquisitionData`.

Reply:
Good point, I will change this. As mentioned above, do you want separate classes for TXRM and TXM data or just one?